FROM ubuntu:18.04

ENV DEBIAN_FRONTEND=noninteractive

RUN apt update && apt install -y iputils-ping rrdtool python3 python3-pip locales tzdata openvpn
COPY ct_shared2/ /
RUN locale-gen en_US.UTF-8
RUN pip3 install -r requirements.txt
RUN echo "export PYTHONIOENCODING=utf-8" >> /etc/profile
ENV PYTHONIOENCODING=utf-8
COPY ./default_locale /etc/default/locale
RUN chmod 0755 /etc/default/locale

ENV LC_ALL=en_US.UTF-8
ENV LANG=en_US.UTF-8
ENV LANGUAGE=en_US:en
RUN ln -fs /usr/share/zoneinfo/Asia/Istanbul /etc/localtime && dpkg-reconfigure -f noninteractive tzdata

CMD ["/bin/bash", "-c", "openvpn --config openvpn.conf --daemon && ./ag_izleme.py start && while true; do sleep 1000; done"]
# The loop was added because the other processes had been working in daemon mode;
# as a result container had been running and being stopped immediately.
# This behaviour shows up for the reason that containers need a process running foreground to be kept running itself.